pkg/steps/multi_stage.go

```diff
@@ -758,17 +762,23 @@ func addCommandScript(name string, pod *coreapi.Pod) {
 	})
 }
 
-func addCliInjector(release string, pod *coreapi.Pod) {
+func addCliInjector(release, pullSpec string, pod *coreapi.Pod) {
```

Comment:
Why not just always pass the full `pullSpec` - that's the only thing it looks like the `release` gets used for. Passing in two options to a method and having it always choose which to use is a code smell